packages/ai/core/generate-image/generate-image.ts

```diff
@@ -133,30 +145,43 @@ Only applicable for HTTP-based providers.
     ),
   );
 
-  // collect result images & warnings
+  // collect result images, warnings, and response metadata
   const images: Array<DefaultGeneratedImage> = [];
   const warnings: Array<ImageGenerationWarning> = [];
-
+  const responses: Array<ImageModelResponseMetadata | undefined> = [];
+  const currentDate = _internal?.currentDate?.() ?? new Date();
   for (const result of results) {
     images.push(
       ...result.images.map(image => new DefaultGeneratedImage({ image })),
     );
     warnings.push(...result.warnings);
+    responses.push({
+      timestamp: currentDate,
+      modelId: model.modelId,
+      headers: result.response?.headers,
+    });
+  }
+
+  if (!images.length) {
+    throw new NoImageGeneratedError({ responses });
   }
 
-  return new DefaultGenerateImageResult({ images, warnings });
+  return new DefaultGenerateImageResult({ images, warnings, responses });
 }
 
 class DefaultGenerateImageResult implements GenerateImageResult {
   readonly images: Array<GeneratedImage>;
   readonly warnings: Array<ImageGenerationWarning>;
+  readonly responses: Array<ImageModelResponseMetadata | undefined>;
```

Comment:
ideally we remove the `undefined` (if we make the information mandatory in the spec)